File: Leetcode/sol3248.py
class Solution:
    def finalPositionOfSnake(self, n: int, commands: List[str]) -> int:

        x, y = 0, 0

        for i in commands:
            match i:
                case "UP":
                    x -= 1
                case "DOWN":
                    x += 1
                case "LEFT":
                    y -= 1
                case "RIGHT":
                    y += 1

        return n * x + y

File: Leetcode/cpp/sol3248.cpp
class Solution {
public:
    int finalPositionOfSnake(int n, vector<string>& commands) {

        int x = 0, y = 0;

        for (string i: commands){
            if(i == "DOWN"){
                x++;
            }else if(i == "UP"){
                x--;
            }else if(i == "LEFT"){
                y--;
            }else{
                y++;
            }
        }

        return n*x+y;
    }
};